rfcs/117-rename-service-to-system-as-the-opposite-to-global.md

```diff
@@ -0,0 +1,47 @@
+---
+author: JinnyYi <github.com/JinnyYi>
+status: draft
+updated_at: 2021-06-21
+---
+
+# GSP-117: Rename Service to System as the Opposite to Global
+
+## Background
+
+The word `Service` has different meanings among our repositories:
+
+- In [go-storage], `Storager` is the main interface for storage service, and `Servicer` is the container for storage. `Storage` and `Service` correspond to the abstraction layers `Storager` and `Servicer` respectively. `ServiceError` and `StorageError` follow the concept.
+- In [specs], `Service Pair` or `Service Info` means the pair or info could only be used in the current service, which is opposite to `Global Pair` or `Global Info`. So we have `Service Level Object Metadata` to `serviceMetadata` in `Object`.
+
+These seem to be confusing for developers and users. So we decide to use `system` as service or system defined information opposite to `global` to fix the confusion on `service pair`, `service metadata`, etc in [Idea: Find a new word to represent Service].
+
+## Proposal
+
+All our repositories should reflect this change.
+
+- Rename `Service Pair` to `System Pair` to represent the input argument for operations in current service, which is opposite to `Global Pair`.
+- Rename `Service Info` to `System Info` to represent the returning information from the current service, which is opposite to `Global Info`.
+- Rename `service Metadata` to `system metadata` to represent system-defined metadata.
+
+## Rationale
+
+N/A
+
+## Compatibility
+
+All API call that used object system metadata could be affected. We could migrate as follows:
+
+- Update fields in `Object` and rename `ObjectMetadata`.

```

Comment:
Maybe we need to add `systemMetadata` first and migration all services, then we remove `serviceMetadata`?

And `GetObjectMetadata` should be marked as deprecated?